pubsub/pulsar/pulsar.go

```diff
@@ -96,38 +110,61 @@ func (p *Pulsar) Subscribe(req pubsub.SubscribeRequest, handler func(msg *pubsub
 		Topic:            req.Topic,
 		SubscriptionName: p.metadata.ConsumerID,
 		Type:             pulsar.Failover,
+		MessageChannel:   channel,
 	}
 
-	options.MessageChannel = channel
 	consumer, err := p.client.Subscribe(options)
 	if err != nil {
 		p.logger.Debugf("Could not subscribe %s", req.Topic)
+		return err
 	}
 
 	go p.ListenMessage(consumer, req.Topic, handler)
 
 	return nil
 }
 
-func (p *Pulsar) ListenMessage(msgs pulsar.Consumer, topic string, handler func(msg *pubsub.NewMessage) error) {
-	for cm := range msgs.Chan() {
-		p.HandleMessage(cm, topic, handler)
+func (p *Pulsar) ListenMessage(consumer pulsar.Consumer, topic string, handler func(msg *pubsub.NewMessage) error) {
+	defer consumer.Close()
+
+	for {
+		select {
+		case msg := <-consumer.Chan():
+			if err := p.HandleMessage(msg, topic, handler); err != nil {
+				p.logger.Errorf("Error processing message")
```

Comment:
There is no detail in this log. And it will print for every error message, should this be a debug log?

Reply:
Good catch.  I added more context on the message here and also filtering out `context.Cancelled`. More work will be done here in the follow up PR regard backoff configuration I mentioned in the description.